### PR TITLE
validate start time on reservation request

### PR DESCRIPTION
**Pull request by erichulburd**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -447,8 +447,13 @@ export function separateCurrentUpcomingReservations(
 }
 
 export function makeAvailabilityRequest(startTime: string, duration: string, latticeName?: string): AvailabilityRequest {
+  const startMoment = convertNaturalDateStringToMoment(startTime);
+  const permitOffset = moment.duration(1, 'minute');
+  if (startMoment < moment().subtract(permitOffset)) {
+    throw new Error(`Must use a date a future to make availability request. Received ${startMoment.toISOString()}.`);
```
Comment (cbosborn) [inline src/utils.ts]: " ... *in the future to make ..."

File: src/utils.ts [hunk 2]
```diff
@@ -447,8 +447,13 @@ export function separateCurrentUpcomingReservations(
 }
 
 export function makeAvailabilityRequest(startTime: string, duration: string, latticeName?: string): AvailabilityRequest {
+  const startMoment = convertNaturalDateStringToMoment(startTime);
+  const permitOffset = moment.duration(1, 'minute');
```
Comment (cbosborn) [inline src/utils.ts:453]: Are you adding 1 minute to ensure that "now" is effectively in the future?

File: test/reserve.test.ts [hunk 3]
```diff
@@ -60,4 +61,14 @@ describe('test the qcs reserve command', () => {
     .it('should call the reserve command and verify output for a multiple returned availabilities', ctx => {
       expect(ctx.stdout).to.equal(pluralOutput);
     });
+
+  test
+    .stdout()
+    .command(['reserve', '-l', latticeName, '-t', '60m', '-n', reservationNotes, '-s', moment().subtract(moment.duration(5, 'minutes')).toISOString()])
+    .catch((err) => {
+      expect(err.message).to.include('Must use a date a future to make availability request');
```
Comment (cbosborn) [inline test/reserve.test.ts]: Same here, "... *in the future ..." unless I'm misreading this.

**Review by cbosborn — CHANGES_REQUESTED** [review]
@erichulburd Also looks good, but hold off for a bit on merging. Curious specifically, what's the purpose of this PR?

**Comment by erichulburd** [comment]
@cbosborn There was a bit of confusion last week with somebody trying to reserve a time in the past and Celena mentioned it'd be nice to get a user friendly error message when that happens.

I can include a force flag to accommodate users when trying to reserve in the past.

**Comment by erichulburd** [comment]
Don't need in this in qcs-cli

**Pull request closed without merging**